src/main/java/org/biouno/unochoice/util/JenkinsUtils.java

```diff
@@ -0,0 +1,293 @@
+              String jarClassPathString = jar.absolutize().toURI().toURL().getPath();
+              classPathList.add(jarClassPathString);
+            }
+          }
+      } 
+      catch(Exception exp) 
+      {
+          logger.log(Level.INFO, "Cannot access path", exp);
+      }
+      config.setClasspathList(classPathList);
+      GroovyShell groovyShell = new GroovyShell(config);
+      
+      for (Entry<String, String> parameter : parameters.entrySet())
+      {
+        groovyShell.setVariable(parameter.getKey(), parameter.getValue());
+      }
+
+      // execute script
+      Object evaluate = groovyShell.evaluate(script);
+
+      return evaluate;
+    }
+    catch (Exception e)
+    {
+      logger.log(Level.SEVERE, "Cannot access class path", e);
+      return null;
+    }
+  }
+
+  /**
+   * Check if a plugin is available.
+   * @param shortName plugin short name
+   * @return {@code true} if plugin is installed
+   */
+  public static boolean isPluginAvailable(String shortName)
+  {
+    Plugin plugin = Hudson.getInstance().getPlugin(shortName);
+    if(plugin == null)
+    {
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Check if the Scriptler plugin is available.
+   * @return {@code true} if Scriptler is installed
+   */
+  public static boolean isScriptlerAvailable()
+  {
+    return JenkinsUtils.isPluginAvailable("scriptler");
+  }
+
+  /**
+   * Get all Scriptler scripts.
+   * @return a set of Scriptler scripts
+   */
+  public static Set<Script> getAllScriptlerScripts()
+  {
+    Set<Script> scripts = ScriptlerConfiguration.getConfiguration().getScripts();
+    return scripts;
+  }
+
+  /**
+   * Find an active node channel for the label of the current project.
+   * @param label label whose nodes to check
+   * @return active node channel or {@code null} if none found
+   */
+  public static VirtualChannel findActiveChannel(Label label)
+  {
+    Iterator<Node> iterator = label.getNodes().iterator();
+    while (iterator.hasNext())
+    {
+      final VirtualChannel channel = iterator.next().getChannel();
+      if (channel != null)
+      {
+        return channel;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Find the label assigned to the current project.
+   * @return {@code null} if the label of the current project cannot be found
+   */
+  @SuppressWarnings("rawtypes")
+  public static Label findProjectLabel(UUID parameterUUID)
+  {
+	AbstractProject project = findCurrentProject(parameterUUID);
+    if (project != null)
+    {
+       return project.getAssignedLabel();
+    }
+    return null;
+  }
+
+  /**
+   * Find the current project.
+   * @return {@code null} if the current project cannot be found
+   */
+  @SuppressWarnings("rawtypes")
+  public static AbstractProject findCurrentProject(UUID parameterUUID)
+  {
+    Jenkins instance = Jenkins.getInstance();
+    if (instance != null)
+    {
+      List<AbstractProject> projects = instance.getAllItems(AbstractProject.class);

```

Comment:
This is possibly quite time consuming, and the plug-in needs to speed up the rendering time (see another PR). Will probably try different approaches before using this one. Large installations (+100 projects) in special, wouldn't be able to use the plug-in I believe. Let's see if we can find another way :-)